addons/product_email_template/models/account_invoice.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+
+from openerp import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def invoice_validate_send_email(self):
+        for invoice in self:
+            # send template only on customer invoice
+            if invoice.type != 'out_invoice':
+                continue
+            # subscribe the partner to the invoice
+            if invoice.partner_id not in invoice.message_partner_ids:
+                invoice.message_subscribe([invoice.partner_id.id])
+            for line in invoice.invoice_line_ids:
+                if line.product_id.email_template_id:
+                    # CLEANME: should define and use a clean API: message_post with a template
+                    Composer = self.env['mail.compose.message'].create({
+                        'model': 'account.invoice',
+                        'res_id': invoice.id,
+                        'template_id': line.product_id.email_template_id.id,
+                        'composition_mode': 'comment',
+                    })
+                    template_values = Composer.onchange_template_id(line.product_id.email_template_id.id, 'comment', 'account.invoice', invoice.id)['value']
+                    template_values['attachment_ids'] = [(4, id) for id in template_values.get('attachment_ids', [])]
```

Comment:
use proper name. like [ [(4, attach) for attach in template_values.get('attachment_ids')]]